plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/SingletonIdentityCacheMapping.java

```diff
@@ -38,7 +38,7 @@ public int hashCode()
         @Override
         public boolean equals(Object obj)
         {
-            return obj instanceof SingletonIdentityCacheMapping;
+            return obj == this;
```

Comment:
```suggestion
            return obj instanceof SingletonIdentityCacheKey;
```

i know it's not copy-paste friendly, but actually 'more correct', should someone manage to create a second instance of the class.

Reply:
>             return obj instanceof SingletonIdentityCacheKey;

This single one-liner was how this PR looked like originally. Are you suggesting we're overcomplicating things? :)